lib/tasks/term_generation/calculate_project_statistics.rake

```diff
@@ -0,0 +1,35 @@
+namespace :term_generation do
+  desc "Calculate statistics for a Term Generation project"
+  task :calculate_project_statistics, [:project_id] => :environment do |_, args|
+    project = TaxonomyProject.find(args.project_id)
+
+    terms = Set.new([])
+    project.taxonomy_todos.done.order(completed_at: :asc).each do |todo|
+      count_of_previously_found_terms = terms.size
+      terms.merge Set.new(todo.terms)
+      todo.update_attribute(:count_of_new_terms, terms.size - count_of_previously_found_terms)
+
+      previous50 = TaxonomyTodo.done
+                                .order(completed_at: :desc)
+                                .where("completed_at <= ?", todo.completed_at)
+                                .limit(50)
```

Comment:
could you make the '50' a constant/parameter? It seems a bit hard-coded like this.